packages/libs/pactjs-generator/src/contract/generation/StringContractDefinition.ts

```diff
@@ -12,10 +12,24 @@ export class StringContractDefinition implements IContractDefinition {
   private _contract: string;
   private _logger: ILogger;
 
-  public constructor(contract: string, logger: ILogger = () => {}) {
+  public constructor(
+    contract: string,
+    namespace: string = '',
+    logger: ILogger = () => {},
+  ) {
     this._contract = contract;
     this._logger = logger;
     this._raw = parser(this._contract, this._logger);
+
+    this._setNamespace(namespace);
+  }
+
+  private _setNamespace(namespace: string): void {
+    if (!namespace) return;
+
+    Object.keys(this._raw).forEach((module) => {
+      this._raw[module].namespace = namespace;
+    });
   }
```

Comment:
Maybe better to leave namespace as optional value. So `namespace: string | undefined`. If it's undefined, it'll take it from the `parser()`. When it's not `undefined`, it'll overwrite whatever comes from the `parser()`